### ISSUE TITLE
[BUG] Remove tags from multiple hypervisor resources but tags on some resources not cleaned

### ISSUE THREAD

**Issue by weizhang1233** [Azure/azure-sdk-for-net#41094]
### Library name and version

id='Azure.ResourceManager',version='1.9.0',

### Describe the bug

I am observing that the tags on specific resources in hypervisor will not be removed as proper if you try to delete those tags from those resources with multiple times call. 

Created multiple virtual machines with network interface, os disk, virtual machine etc resources. (3 or more virtual machines need to be introduced). 

Try to remove tags from disk one by one (as part of loop, you can delete tags with same way in loop from virtual machine, network interfaces etc) with no specific time interval setup in this clean tags loop. 

` try
 {
     var tags= client.GetGenericResource(identifierinstance).GetTagResource();
     var pr= new TagResourcePatch()
     {
         PatchMode = TagPatchMode.Delete,
     };
     //setup tags key value to pr (TagResourcePatch())
     return (await tags.UpdateAsync(WaitUntil.Completed, patchReq).ConfigureAwait(false)).Value.Data;
 }
 catch (Exception e)
 {
     ////
 }`
 Maybe you will find an exception due to issue https://github.com/Azure/azure-sdk-for-net/issues/41063 or not find any exception with everything running as proper. Them you go to hypervisor side, found tags not remove to specific one of mulitple disks. Other resources like network interface, virtual machine, other disk etc tags have been cleaned successfully. 

Go to hypervisor, resource group, actitivy log, to that disk, looks like no alarm or error from hypervisor side log, 
![image](https://github.com/Azure/azure-sdk-for-net/assets/155944395/2d4bf0d7-bc56-4de6-aab2-686a3a522710)

![image](https://github.com/Azure/azure-sdk-for-net/assets/155944395/436a77dc-9582-4280-8627-e525b1aec8b5)

![image](https://github.com/Azure/azure-sdk-for-net/assets/155944395/1135510a-25e8-4176-82a8-a44b84973821)

### Expected behavior

All identicated tags will be removed from those resources. 

### Actual behavior

tags on specific os disk will not be cleaned as proper. 

### Reproduction Steps

Please refer to description in "Describe the bug" you will find de steps to reproduce it. 

### Environment

windows

.NET SDK:
 Version:           8.0.100

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by ArthurMa1978** [comment]
Thank you @weizhang1233 for reaching us, @HarveyLink  will follow up your question.

**Comment by HarveyLink** [comment]
Hi @weizhang1233 , We can't repro your issue with following code:
```
async Task<TagResourceData> UpdateTagResourceAsync(ArmClient client, ResourceIdentifier id)
{
    var tags = client.GetGenericResource(id).GetTagResource();
    var patchReq = new TagResourcePatch()
    {
        PatchMode = TagPatchMode.Replace,
        TagValues =
        {
            {"key1", "value1"},
            {"key2", "value2"},
        }
    };
    return (await tags.UpdateAsync(WaitUntil.Completed, patchReq)).Value.Data;
}
```
Please make sure there is no other tags in the disk resource, the remove operation could only apply on the matched key/value.
If possible, please capture and attach the request/response payload, it could help us to identify the issue.

**Comment by weizhang1233** [comment]
@jsquire Any updating? Thanks

**Comment by jsquire** [comment]
@weizhang1233 : I have no insight into the current state.  As mentioned earlier, @HarveyLink is the engineer assigned to assist.  Any updates will come from them.

**Comment by HarveyLink** [comment]
> n 3 sets of resources( with VM, vnet, ipaddress, networkinterface, nsg, disk). The tags could be set and removed as we want.

I did some research today, do found some instersting behavior.
While operate the tag value of `Disk`, it seems we need to add delay(`Thread.Sleep()`) between update and remove methods to make both operations work. Otherwise, it won't do the update and leave the remove not working. 
The delay should be more than 7 seconds, this is the least time I found so far.
In conclusion, based on the above results, I believe this should be a service issue.

**Comment by weizhang1233** [comment]
@HarveyLink Saw you ad needs-author-feedback label. What's the the information you need? 
What I am doing is introduce a TagResourcePatch with patch mode as TagPatchMode.Delete, then add some tags to  TagResourcePatch.TagValues, which will be deleted from hypervisor resource. Run updateAsync to to execute removing. 

What do you mean add delay between update and remove methods?  I tried to delete tags on one resource successfully. But if you try to delete tags from multiple virtual machines (Include resources disk, nic etc to each individule), tags on some specific resource not moved as proper. 

>>>
"I did some research today, do found some interesting behavior.
While operate the tag value of Disk, it seems we need to add delay(Thread.Sleep()) between update and remove methods to make both operations work. Otherwise, it won't do the update and leave the remove not working.
The delay should be more than 7 seconds, this is the least time I found so far.
In conclusion, based on the above results, I believe this should be a service issue."
